test/rules/no-mocha-arrows.js

```diff
@@ -2,49 +2,90 @@
+es6RuleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
 
     valid: [
         'it()',
         'it(function() { assert(something, false); })',
         'it("should be false", function() { assert(something, false); })',
-        {
-            // In this example, `it` is not a global.
-            code: 'function it () {}; it(() => { console.log("okay") })',
-            parserOptions: { ecmaVersion: 6 }
-        },
-        'it.only()'
+         // In this example, `it` is not a global.
+        'function it () {}; it(() => { console.log("okay") })',
+        'it.only()',
+        'it(function(done) { assert(something, false); done(); })',
+        'it(function*() { assert(something, false) })'
     ],
 
     invalid: [
         {
             code: 'it(() => { assert(something, false); })',
-            parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: errors,
+            output: 'it(function () { assert(something, false); })'
         },
         {
             code: 'it(() => { assert(something, false); })',
             globals: [ 'it' ],
-            parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: errors,
+            output: 'it(function () { assert(something, false); })'
         },
         {
             code: 'it(() => assert(something, false))',
-            parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: errors,
+            output: 'it(function () { return assert(something, false); })'
         },
         {
             code: 'it("should be false", () => { assert(something, false); })',
-            parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: errors,
+            output: 'it("should be false", function () { assert(something, false); })'
         },
         {
             code: 'it.only(() => { assert(something, false); })',
-            parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: 'Do not pass arrow functions to it.only()', column: 1, line: 1 } ]
+            errors: [ { message: 'Do not pass arrow functions to it.only()', column: 1, line: 1 } ],
+            output: 'it.only(function () { assert(something, false); })'
+        },
+        {
+            code: 'it((done) => { assert(something, false); })',
+            errors: errors,
+            output: 'it(function (done) { assert(something, false); })'
+        },
+        {
+            code: 'it("should be false", () => {\n assert(something, false);\n})',
+            errors: errors,
+            output: 'it("should be false", function () {\n assert(something, false);\n})'
+        }
+    ]
+
+});
+
+es2017RuleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
```

Comment:
I feel it's enough to use test runner using es2016 parser for every tests.  I'll make one test runner for tests.